wasm/src/decoder/type_section.rs

```diff
@@ -0,0 +1,19 @@
+use super::common;
+use super::util;
+use crate::error::Error;
+use crate::error::Result;
+
+const FN_SIGNATURE: u8 = common::ValueKind::Func as u8;
+
+pub fn decode_type_section<R: std::io::Read>(reader: &mut R) -> Result<common::FnType> {
```

Comment:
Does this allow modules that lack functions?
(I don't know how this part is being used)

Reply:
yes. This only get's called when modules do have a type_section and thus a function saction